### PR TITLE
Fix Issue 20406 - Copy constructor requires default constructor

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

**Comment by veelo** [comment]
Kudos for your quick reaction, Razvan!

**Comment by RazvanN7** [comment]
Thanks! After all, it is my baby..

**Comment by veelo** [comment]
> Thanks! After all, it is my baby..

She'll grow up to be a valuable member of the family :-)

File: src/dmd/mtype.d [hunk 1]
```diff
@@ -4736,6 +4736,7 @@ extern (C++) final class TypeFunction : TypeNext
                              * copytmp.__copyCtor(arg);
                              */
                             auto tmp = new VarDeclaration(arg.loc, tprm, Identifier.generateId("__copytmp"), null);
```
Comment (SSoulaimane) [inline src/dmd/mtype.d]: There is a more lightweight technique to probe the copy constructor but maybe leave refactoring for a separate PR.
See:
https://github.com/dlang/dmd/blob/664f07c08d4331b53e6fcb419a070c808dc4e71d/src/dmd/clone.d#L103
https://github.com/dlang/dmd/blob/664f07c08d4331b53e6fcb419a070c808dc4e71d/src/dmd/clone.d#L113

**Review by SSoulaimane — APPROVED** [review]
LG

File: src/dmd/mtype.d (hunk 1 above)
Comment (RazvanN7) [reply src/dmd/mtype.d]: Yep. Even the temporary generation can be substituted with a simple function call (copyToTemp), but I didn't want to mix any refactorizations in.

**Review by thewilsonator — APPROVED** [review]
